python/semantic_kernel/agents/agent.py

```diff
@@ -32,7 +33,7 @@ class Agent(KernelBaseModel):
 
     id: str = Field(default_factory=lambda: str(uuid.uuid4()))
     description: str | None = None
-    name: str = Field(default_factory=lambda: f"agent_{generate_random_ascii_name()}")
+    name: str = Field(default_factory=lambda: f"agent_{generate_random_ascii_name()}", pattern=AGENT_NAME_REGEX)
```

Comment:
I've started noticing (and making changes) that using this syntax works better and is also now recommended by pydantic, for all but the default_factory:
```suggestion
    name: Annotated[str, Field(pattern=AGENT_NAME_REGEX)] = Field(default_factory=lambda: f"agent_{generate_random_ascii_name()}")
```